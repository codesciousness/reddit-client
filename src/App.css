/* Universal Styles */

* , *::before, *::after {
  margin: 0;
  border: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  font-size: 16px;
}

body {
  font-size: 1.5rem;
}

.App {
  min-height: 100vh;
  background: lightgray;
}

/* Title */

.App__title {
  height: 7vh;
  line-height: 7vh;
  font-size: 2rem;
  background: black;
  opacity: 0.7;
}

/* Header */

.App__header {
  min-height: 70vh;
  font-size: 1rem;
  text-align: center;
  color: white;
  background-image: linear-gradient(180deg, rgba(0, 0, 0, 0.1), rgba(0, 0, 0, 0.9)), url(./resources/images/ben-wicks-Ia-qPL-HQdA-unsplash.jpg);
  background-size: cover;
  border-top: 3px solid gray;
  border-bottom: 3px solid gray;
  border-image: linear-gradient(to right, rgb(255, 2, 108), orange, yellow, green, blue, purple) 1;
  position: relative;
}