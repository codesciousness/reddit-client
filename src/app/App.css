/* Universal Styles */

* , *::before, *::after {
  margin: 0;
  border: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  font-size: 16px;
}

body {
  font-family: 'Asap', sans-serif;
  font-size: 1rem;
}

.App {
  min-height: 100vh;
  background: whitesmoke;
  padding-bottom: 3rem;
}

.App__navbar {
  height: 7vh;
  line-height: 7vh;
  font-size: 2rem;
  background: black;
  opacity: 0.7;
  position: relative;
}

.App__navbar__icon {
  color: white;
  display: inline-block;
  position: absolute;
  left: 0;
  padding: 0.75rem 0.5rem 0.5rem 1rem;
}

/* Titles */

.App__title {
  color: white;
  display: inline-block;
  font-size: 2rem;
}

.App__subtitle {
  font-size: 2rem;
  font-weight: normal;
  margin: 3rem auto 2rem auto;
}

/* Header */

.App__header {
  min-height: 70vh;
  text-align: center;
  color: white;
  background-image: linear-gradient(180deg, rgba(0, 0, 0, 0.1), rgba(0, 0, 0, 0.9)), url(../resources/images/ben-wicks-Ia-qPL-HQdA-unsplash.jpg);
  background-size: cover;
  border-top: 3px solid gray;
  border-bottom: 3px solid gray;
  border-image: linear-gradient(to right, rgb(255, 2, 108), orange, yellow, green, blue, purple) 1;
  position: relative;
}

.App__trending, .App__latest {
  width: 80%;
  margin: 3rem auto;
}

.App__divider__container {
  width: 90%;
  margin: 2rem auto;
  position: relative;
}

.App__divider__container hr {
  border-bottom: 1px solid black;
}

.App__divider {
  display: inline-block;
  position: absolute;
  top: -50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-weight: bold;
  text-align: center;
  text-transform: uppercase;
  color: white;
  background: linear-gradient(to right, rgb(255, 2, 108), purple);
  padding: 0.25rem;
}