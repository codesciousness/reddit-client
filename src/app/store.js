import { configureStore } from '@reduxjs/toolkit';
import searchTermReducer from '../features/searchTerm/searchTermSlice';
import filterReducer from '../features/filter/filterSlice';

export default configureStore({
  reducer: {
    searchTerm: searchTermReducer,
    filter: filterReducer
  },
});