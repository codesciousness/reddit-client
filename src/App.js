import React from 'react';
//import logo from './logo.svg';
import './App.css';
import Filters from './features/filters/Filters';
import SearchBar from './features/searchbar/SearchBar';

function App() {
  return (
    <div className="App">
      <header className="App__header">
        <h1 className="App__title">Reddit Science</h1>
        <Filters />
      </header>
      <SearchBar />
    </div>
  );
}

export default App;