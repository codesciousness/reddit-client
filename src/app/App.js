import React from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
//import logo from './logo.svg';
import './App.css';
import FilterList from '../components/filterList/FilterList';
import Home from '../components/home/Home';
import PostDetail from '../components/postDetail/PostDetail';

function App() {
  return (
    <Router>
      <div className="App">
        <header className="App__header">
          <nav className="App__navbar">
            <Link to='/'><i className="fas fa-home App__navbar__icon"></i></Link>
            <Link to='/'><h1 className="App__title">Reddit Science</h1></Link>
          </nav>
          <h2 className="App__subtitle">Your Daily Dose of Science</h2>
          <FilterList />
        </header>
        <Switch>
          <Route exact path='/' component={Home} />
          <Route path='/posts/:flair/:id' component={PostDetail} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;