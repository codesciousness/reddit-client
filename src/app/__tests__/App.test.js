import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import store from '../store';
import App from '../App';

describe('<App />', () => {
  test('App component is defined', () => {
    expect(<App />).toBeDefined();
  });
  test('renders correctly with initial state from Redux store', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(screen.getByText('Reddit Science')).toBeInTheDocument();
  });
  test('matches the snapshot', () => {
    const component = create(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});