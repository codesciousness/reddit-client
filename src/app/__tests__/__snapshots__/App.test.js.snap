// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App__header"
  >
    <nav
      className="App__navbar"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-home App__navbar__icon"
        />
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="App__title"
        >
          Reddit Science
        </h1>
      </a>
    </nav>
    <section
      className="FilterList"
    >
      <h2
        className="FilterList__title"
      >
        Your Daily Dose of Science
      </h2>
      <p
        className="FilterList__prompt"
      >
        What interests you?
      </p>
      <ul
        className="FilterList__container"
      >
        <li>
          <button
            className="Filter__button"
            id="AnimalScience"
            onClick={[Function]}
          >
            Animal Science
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Anthropology"
            onClick={[Function]}
          >
            Anthropology
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Astronomy"
            onClick={[Function]}
          >
            Astronomy
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Biology"
            onClick={[Function]}
          >
            Biology
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Cancer"
            onClick={[Function]}
          >
            Cancer
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Chemistry"
            onClick={[Function]}
          >
            Chemistry
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="ComputerScience"
            onClick={[Function]}
          >
            Computer Science
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="EarthScience"
            onClick={[Function]}
          >
            Earth Science
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Economics"
            onClick={[Function]}
          >
            Economics
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Engineering"
            onClick={[Function]}
          >
            Engineering
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Environment"
            onClick={[Function]}
          >
            Environment
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Epidemiology"
            onClick={[Function]}
          >
            Epidemiology
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Geology"
            onClick={[Function]}
          >
            Geology
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Health"
            onClick={[Function]}
          >
            Health
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Mathematics"
            onClick={[Function]}
          >
            Mathematics
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Medicine"
            onClick={[Function]}
          >
            Medicine
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Nanoscience"
            onClick={[Function]}
          >
            Nanoscience
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Neuroscience"
            onClick={[Function]}
          >
            Neuroscience
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Paleontology"
            onClick={[Function]}
          >
            Paleontology
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Physics"
            onClick={[Function]}
          >
            Physics
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="Psychology"
            onClick={[Function]}
          >
            Psychology
          </button>
        </li>
        <li>
          <button
            className="Filter__button"
            id="SocialScience"
            onClick={[Function]}
          >
            Social Science
          </button>
        </li>
      </ul>
    </section>
  </header>
  <main>
    <form
      className="SearchTerm"
    >
      <i
        className="fas fa-search fa-lg SearchTerm__icon"
      />
      <input
        className="SearchTerm__input"
        onChange={[Function]}
        placeholder="Search..."
        value=""
      />
      
    </form>
    <section
      className="Home__trending"
    >
      <div
        className="Home__divider__container"
      >
        <hr />
        <div
          className="Home__divider"
        >
          Trending
        </div>
      </div>
      <section
        className="Posts__trending"
      >
        <ul>
          <li>
            <section
              className="PostLoader__trending"
            >
              <div
                className="PostLoader__trending__content"
              >
                <h2
                  className="PostLoader__trending__flair"
                >
                  Post flair name
                </h2>
                <h3
                  className="PostLoader__trending__title"
                >
                  Short title of the article...
                </h3>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__trending"
            >
              <div
                className="PostLoader__trending__content"
              >
                <h2
                  className="PostLoader__trending__flair"
                >
                  Post flair name
                </h2>
                <h3
                  className="PostLoader__trending__title"
                >
                  Short title of the article...
                </h3>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__trending"
            >
              <div
                className="PostLoader__trending__content"
              >
                <h2
                  className="PostLoader__trending__flair"
                >
                  Post flair name
                </h2>
                <h3
                  className="PostLoader__trending__title"
                >
                  Short title of the article...
                </h3>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__trending"
            >
              <div
                className="PostLoader__trending__content"
              >
                <h2
                  className="PostLoader__trending__flair"
                >
                  Post flair name
                </h2>
                <h3
                  className="PostLoader__trending__title"
                >
                  Short title of the article...
                </h3>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__trending"
            >
              <div
                className="PostLoader__trending__content"
              >
                <h2
                  className="PostLoader__trending__flair"
                >
                  Post flair name
                </h2>
                <h3
                  className="PostLoader__trending__title"
                >
                  Short title of the article...
                </h3>
              </div>
            </section>
          </li>
        </ul>
      </section>
    </section>
    <section
      className="Home__latest"
    >
      <div
        className="Home__divider__container"
      >
        <hr />
        <div
          className="Home__divider"
        >
          Latest
        </div>
      </div>
      <section
        className="Posts__latest"
      >
        <ul>
          <li>
            <section
              className="PostLoader__latest"
            >
              <div
                className="PostLoader__latest__image"
              />
              <div
                className="PostLoader__latest__content"
              >
                <p
                  className="PostLoader__latest__flair"
                >
                  Post flair name
                </p>
                <h2
                  className="PostLoader__latest__title"
                >
                  Complete title of the article
                </h2>
                <button
                  className="PostLoader__latest__button"
                >
                  Read more
                </button>
                <p
                  className="PostLoader__latest__detail"
                >
                  <span
                    className="PostLoader__latest__author"
                  >
                    Posted by 
                    <strong>
                      Post author
                    </strong>
                  </span>
                  <span
                    className="PostLoader__latest__time"
                  >
                    Post time
                  </span>
                  <span
                    className="PostLoader__latest__comments"
                  >
                    <i
                      className="far fa-comment-alt"
                    />
                     
                    <strong>
                      Comment#
                    </strong>
                  </span>
                </p>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__latest"
            >
              <div
                className="PostLoader__latest__image"
              />
              <div
                className="PostLoader__latest__content"
              >
                <p
                  className="PostLoader__latest__flair"
                >
                  Post flair name
                </p>
                <h2
                  className="PostLoader__latest__title"
                >
                  Complete title of the article
                </h2>
                <button
                  className="PostLoader__latest__button"
                >
                  Read more
                </button>
                <p
                  className="PostLoader__latest__detail"
                >
                  <span
                    className="PostLoader__latest__author"
                  >
                    Posted by 
                    <strong>
                      Post author
                    </strong>
                  </span>
                  <span
                    className="PostLoader__latest__time"
                  >
                    Post time
                  </span>
                  <span
                    className="PostLoader__latest__comments"
                  >
                    <i
                      className="far fa-comment-alt"
                    />
                     
                    <strong>
                      Comment#
                    </strong>
                  </span>
                </p>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__latest"
            >
              <div
                className="PostLoader__latest__image"
              />
              <div
                className="PostLoader__latest__content"
              >
                <p
                  className="PostLoader__latest__flair"
                >
                  Post flair name
                </p>
                <h2
                  className="PostLoader__latest__title"
                >
                  Complete title of the article
                </h2>
                <button
                  className="PostLoader__latest__button"
                >
                  Read more
                </button>
                <p
                  className="PostLoader__latest__detail"
                >
                  <span
                    className="PostLoader__latest__author"
                  >
                    Posted by 
                    <strong>
                      Post author
                    </strong>
                  </span>
                  <span
                    className="PostLoader__latest__time"
                  >
                    Post time
                  </span>
                  <span
                    className="PostLoader__latest__comments"
                  >
                    <i
                      className="far fa-comment-alt"
                    />
                     
                    <strong>
                      Comment#
                    </strong>
                  </span>
                </p>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__latest"
            >
              <div
                className="PostLoader__latest__image"
              />
              <div
                className="PostLoader__latest__content"
              >
                <p
                  className="PostLoader__latest__flair"
                >
                  Post flair name
                </p>
                <h2
                  className="PostLoader__latest__title"
                >
                  Complete title of the article
                </h2>
                <button
                  className="PostLoader__latest__button"
                >
                  Read more
                </button>
                <p
                  className="PostLoader__latest__detail"
                >
                  <span
                    className="PostLoader__latest__author"
                  >
                    Posted by 
                    <strong>
                      Post author
                    </strong>
                  </span>
                  <span
                    className="PostLoader__latest__time"
                  >
                    Post time
                  </span>
                  <span
                    className="PostLoader__latest__comments"
                  >
                    <i
                      className="far fa-comment-alt"
                    />
                     
                    <strong>
                      Comment#
                    </strong>
                  </span>
                </p>
              </div>
            </section>
          </li>
          <li>
            <section
              className="PostLoader__latest"
            >
              <div
                className="PostLoader__latest__image"
              />
              <div
                className="PostLoader__latest__content"
              >
                <p
                  className="PostLoader__latest__flair"
                >
                  Post flair name
                </p>
                <h2
                  className="PostLoader__latest__title"
                >
                  Complete title of the article
                </h2>
                <button
                  className="PostLoader__latest__button"
                >
                  Read more
                </button>
                <p
                  className="PostLoader__latest__detail"
                >
                  <span
                    className="PostLoader__latest__author"
                  >
                    Posted by 
                    <strong>
                      Post author
                    </strong>
                  </span>
                  <span
                    className="PostLoader__latest__time"
                  >
                    Post time
                  </span>
                  <span
                    className="PostLoader__latest__comments"
                  >
                    <i
                      className="far fa-comment-alt"
                    />
                     
                    <strong>
                      Comment#
                    </strong>
                  </span>
                </p>
              </div>
            </section>
          </li>
        </ul>
      </section>
    </section>
  </main>
</div>
`;
